### ISSUE TITLE
实验效果问题

### ISSUE THREAD

**Issue by SunJiaYuuuuu** [zhouhaoyi/Informer2020#340]
![微信截图_20220613154436](https://user-images.githubusercontent.com/95565810/173304643-e569219b-187f-44fe-8011-9a309e19c0eb.png)

您好，这是我复现代码时将预测值和真实值进行可视化好得到的效果图。
我使用了你们提供的ETTm1数据集，时间间隔设为15t，采用scripts中的参数，--features S --seq_len 384 --label_len 384 --pred_len 288 --e_layers 2 --d_layers 1 。
得到的MSE与MAE为mse:0.25330454111099243, mae:0.4220234155654907
这个mse是较低的，但可视化效果图有点不尽人意。。

**Comment by LCZ2815** [comment]
> ![微信截图_20220613154436](https://user-images.githubusercontent.com/95565810/173304643-e569219b-187f-44fe-8011-9a309e19c0eb.png)
> 
> 您好，这是我复现代码时将预测值和真实值进行可视化好得到的效果图。 我使用了你们提供的ETTm1数据集，时间间隔设为15t，采用scripts中的参数，--features S --seq_len 384 --label_len 384 --pred_len 288 --e_layers 2 --d_layers 1 。 得到的MSE与MAE为mse:0.25330454111099243, mae:0.4220234155654907 这个mse是较低的，但可视化效果图有点不尽人意。。

您好，请问您的效果调好了么？主要是调节哪些参数呢？

**Comment by MountVoom** [comment]
> 你好，我想问下，我照着源码跑出来，想把预测值与真实值可视化，但是只能得到pre_len=24个数字的值，怎么将测试集可视化

您好，测试集的结果会被默认保存在results目录里true.npy和pred.npy中，具体可以参考exp_informer.py中的test函数。

**Comment by MountVoom** [comment]
您好，我也采取了您的参数进行实验，得到的结果是这样的：
![res](https://user-images.githubusercontent.com/33650160/188049709-f76d5c6e-0966-456a-adab-4a56be040de1.png)

我的详细参数是：Namespace(activation='gelu', attn='prob', batch_size=32, c_out=1, checkpoints='./checkpoints/', cols=None, d_ff=2048, d_layers=1, d_model=512, data='ETTm1', data_path='ETTm1.csv', dec_in=1, des='test', detail_freq='15t', devices='0,1,2,3', distil=True, do_predict=False, dropout=0.05, e_layers=2, embed='timeF', enc_in=1, factor=5, features='S', freq='t', gpu=0, inverse=False, itr=1, label_len=384, learning_rate=0.0001, loss='mse', lradj='type1', mix=True, model='informer', n_heads=8, num_workers=0, output_attention=False, padding=0, patience=3, pred_len=288, root_path='./data/ETT/', s_layers=[3, 2, 1], seq_len=384, target='OT', train_epochs=6, use_amp=False, use_gpu=True, use_multi_gpu=False)

得到的MSE和MAE为mse:0.2540646195411682, mae:0.4321681559085846。

可以分享一下您的详细参数设置吗？

**Comment by recha0** [comment]
> 您好，我也采取了您的参数进行实验，得到的结果是这样的： ![res](https://user-images.githubusercontent.com/33650160/188049709-f76d5c6e-0966-456a-adab-4a56be040de1.png)
> 
> 我的详细参数是：Namespace(activation='gelu', attn='prob', batch_size=32, c_out=1, checkpoints='./checkpoints/', cols=None, d_ff=2048, d_layers=1, d_model=512, data='ETTm1', data_path='ETTm1.csv', dec_in=1, des='test', detail_freq='15t', devices='0,1,2,3', distil=True, do_predict=False, dropout=0.05, e_layers=2, embed='timeF', enc_in=1, factor=5, features='S', freq='t', gpu=0, inverse=False, itr=1, label_len=384, learning_rate=0.0001, loss='mse', lradj='type1', mix=True, model='informer', n_heads=8, num_workers=0, output_attention=False, padding=0, patience=3, pred_len=288, root_path='./data/ETT/', s_layers=[3, 2, 1], seq_len=384, target='OT', train_epochs=6, use_amp=False, use_gpu=True, use_multi_gpu=False)
> 
> 得到的MSE和MAE为mse:0.2540646195411682, mae:0.4321681559085846。
> 
> 可以分享一下您的详细参数设置吗？

您好，我在RTX3060的win上按照您给的参数跑了两遍，感觉效果不太好，是我设置的哪里有问题吗？
![image](https://user-images.githubusercontent.com/46513622/204329853-e458cf98-4c91-46ea-acf6-782e64c51f19.png)

这是我的参数：


Args in experiment:
Namespace(model='informer', data='ETTm1', root_path='./data/ETT/', data_path='ETTm1.csv', features='S', target='OT', freq='t', checkpoints='./checkpoints/', seq_len=384, label_len=384, pred_len=288, enc_in=1, dec_in=1, c_out=1, d_model=512, n_heads=8, e_layers=2, d_layers=1, s_layers=[3, 2, 1], d_ff=2048, factor=5, padding=0, distil=True, dropout=0.05, attn='prob', embed='timeF', activation='gelu', output_attention=False, do_predict=False, mix=True, cols=None, num_workers=0, itr=1, train_epochs=6, batch_size=32, patience=3, learning_rate=0.0001, des='test', loss='mse', lradj='type1', use_amp=False, inverse=False, use_gpu=True, gpu=0, use_multi_gpu=False, devices='0,1,2,3', detail_freq='15t')
Use GPU: cuda:0
>>>>>>>start training : informer_ETTm1_ftS_sl384_ll384_pl288_dm512_nh8_el2_dl1_df2048_atprob_fc5_ebtimeF_dtTrue_mxTrue_test_0>>>>>>>>>>>>>>>>>>>>>>>>>>
train 33889
val 11233
test 11233
>>>>>>>testing : informer_ETTm1_ftS_sl384_ll384_pl288_dm512_nh8_el2_dl1_df2048_atprob_fc5_ebtimeF_dtTrue_mxTrue_test_0<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<
test 11233
test shape: (351, 32, 288, 1) (351, 32, 288, 1)
test shape: (11232, 288, 1) (11232, 288, 1)
mse:0.2287609875202179, mae:0.41189518570899963

**Comment by recha0** [comment]
问一下我预测60个，输出维度为1，（1，11872，60，1）这个是输出的维度，那么我可视化输出（0，x，：，0）是最后的结果是么？x要取几

**Comment by zboboj** [comment]
> 
你好，请问一下，代码输出的结果都是（-1，1）之间，你这个图中预测值是什么值，怎么得出的？谢谢

**Comment by MountVoom** [comment]
您好，您只需要使用--inverse反归一化即可，(-1, 1)是归一化之后的结果

**Comment by zboboj** [comment]
> 您好，您只需要使用--inverse反归一化即可，(-1, 1)是归一化之后的结果
万分感谢您的回答，用自己的数据去跑，预测5个点，遇到2个问题非常不理解，希望您帮忙看看，谢谢。
以下是自己数据的参数设置：
Namespace(model='informer', data='sh', root_path='./data/ETT/', data_path='sh.csv', features='M', target='close', freq='b', checkpoints='./checkpoints/', seq_len=480, label_len=240, pred_len=5, enc_in=5, dec_in=5, c_out=5, d_model=512, n_heads=8, e_layers=2, d_layers=1, s_layers=[3, 2, 1], d_ff=2048, factor=5, padding=0, distil=True, dropout=0.05, attn='prob', embed='timeF', activation='gelu', output_attention=False, do_predict=True, mix=True, cols=None, num_workers=0, itr=2, train_epochs=6, batch_size=32, patience=3, learning_rate=0.0001, des='test', loss='mse', lradj='type1', use_amp=False, inverse=True, use_gpu=False, gpu=1, use_multi_gpu=False, devices='0,1,2,3', detail_freq='b')
希
问题1：样本数据有6000多行，输出的结果true.npy为什么是样本数据4535-4539行，预测结果pred.npy的5个点就是对4535-4539这5行的预测吗？
True.npy
![image](https://user-images.githubusercontent.com/34063893/209941437-2f8f2b65-f098-4f82-b2ae-e778c29a82ca.png)

样本数据：
![image](https://user-images.githubusercontent.com/34063893/209941452-a7f1165e-f490-4bc5-abe4-b08a872069e0.png)

问题2：样本数据的最新日期是“2022/12/26”，如何预测“2022/12/26”5天后的数据，该怎么样设置？

**Comment by MountVoom** [comment]
您好，可以参考一下dataloader里对训练集测试集的划分，ture.npy和pred.npy是测试集的真实值和预测结果。对未来进行预测的结果保存在real_prediction.npy中，具体可以参考exp_informer.py里的predict函数。想要预测五天后的数据，可以将pred_len设置的长一些。

**Comment by zboboj** [comment]
> 您好，可以参考一下dataloader里对训练集测试集的划分，ture.npy和pred.npy是测试集的真实值和预测结果。对未来进行预测的结果保存在real_prediction.npy中，具体可以参考exp_informer.py里的predict函数。想要预测五天后的数据，可以将pred_len设置的长一些。

感谢！！！！
